io/genbank/multimap.go

```diff
@@ -0,0 +1,62 @@
+/*
+This file provides utilities for working with a MultiMap,
+which is simply a map which can store multiple values for a single key instead
+of the usual one.
+
+Useful for when we expect to encounter repeated keys but we want to keep all pairs,
+not just the latest one, while preserving O(1) time lookup cost.
+Does not make uniqueness quarantees for key value pairs.
+This may end up being useful for other parsers which allow for repeated keys, in which case
+this should be made into its own module.
+*/
+package genbank
+
+// defined as a simple type alias over a map of slices
+// while not ideal (eg computing total number of items takes O(N))
+// this has the advantage of being compatible with json.Marshal, cmp.Diff,
+// pretty printing, and bracket indexing out of the box.
+type MultiMap[K, V comparable] map[K][]V
+
+// create a new empty multimap
+func NewMultiMap[K, V comparable]() MultiMap[K, V] {
+	return make(map[K][]V)
+}
```

Comment:
Pretty sure the docstrings are supposed to start with the name of the function, or am I wrong there?

Reply:
You are 100% right